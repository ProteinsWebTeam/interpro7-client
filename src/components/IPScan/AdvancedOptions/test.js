// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { AdvancedOptions } from '.';

const renderer = new ShallowRenderer();

describe('<AdvancedOptions />', () => {
  test('should render', () => {
    renderer.render(
      <AdvancedOptions
        data={{
          loading: false,
          payload: {
            description:
              'A number of different protein sequence applications are launched,' +
              ' These applications search against specific databases and have preconfigured cut off ' +
              'thresholds.',
            name: 'Applications',
            type: 'STRING',
            values: {
              values: [],
            },
          },
          ok: true,
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
