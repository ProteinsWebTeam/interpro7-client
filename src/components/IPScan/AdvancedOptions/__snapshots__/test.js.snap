// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedOptions /> should render 1`] = `
<section>
  <details
    className="option-style"
  >
    <summary>
      <span
        className="triangle"
      >
        â–º
      </span>
      <span>
        Advanced options
      </span>
    </summary>
    <fieldset
      className="new-fieldset"
    >
      <legend>
        Job configuration
      </legend>
      <label
        style={
          {
            "marginBottom": "1rem",
          }
        }
      >
        <ToggleSwitch
          id="seqtype"
          label="Sequence type:"
          name="seqtype"
          offValue="protein (amino acids)"
          onValue="DNA/RNA (nucleotides)"
          size="small"
          switchCond={false}
          width="8rem"
        />
      </label>
      <br />
      <label
        style={
          {
            "marginBottom": "1rem",
          }
        }
      >
        <ToggleSwitch
          id="stay"
          label="Create another job after this one:"
          name="stay"
          size="small"
          switchCond={false}
        />
      </label>
      <label
        className="new-input-group"
      >
        <span
          className="new-input-group-label"
        >
          Job title
        </span>
        <input
          className="input-group-field"
          name="local-title"
          placeholder="Give this job a local title (only visible on this browser)"
          type="text"
        />
      </label>
    </fieldset>
    <fieldset
      className="new-fieldset"
    >
      <legend>
        Applications
      </legend>
      <p>
        A number of different protein sequence applications are launched, These applications search against specific databases and have preconfigured cut off thresholds.
      </p>
      <div
        className="button-group line-with-buttons"
      >
        <ForwardRef
          onClick={[Function]}
          type="tertiary"
        >
          Select all
        </ForwardRef>
        <ForwardRef
          onClick={[Function]}
          type="tertiary"
        >
          Unselect all
        </ForwardRef>
      </div>
      <fieldset
        className="new-fieldset"
      >
        <legend>
          Member databases
        </legend>
        <fieldset
          className="new-fieldset"
        >
          <legend>
            Families, domains, sites & repeats
          </legend>
        </fieldset>
        <fieldset
          className="new-fieldset"
        >
          <legend>
            Structural domains
          </legend>
        </fieldset>
      </fieldset>
      <fieldset
        className="new-fieldset"
      >
        <legend>
          Other sequence features
        </legend>
      </fieldset>
    </fieldset>
  </details>
</section>
`;
